### PR TITLE
Softmax Op Migration to TMP Infra

### PR DESCRIPTION
**Pull request by cfjchu**
### Ticket
Link to Github Issue

### Problem description
Migrating softmax op to the new TMP infra. I wanted to go through the migration process myself to see what is involved.

### What's changed
- Softmax operation now migrated to new TMP infra

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp [hunk 1]
```diff
@@ -26,26 +26,34 @@ ttnn::Tensor ExecuteAttentionSoftmax<in_place>::invoke(
...
         if (not attention_mask.has_value()) {
             auto output_tensor = ttnn::multiply(input_tensor, head_size);
-            return ttnn::operations::normalization::softmax(output_tensor, memory_config.value_or(input_tensor.memory_config()));
+            return ttnn::prim::softmax(
+                ttnn::operations::normalization::SoftmaxDeviceOperation::operation_attributes_t{
+                    .memory_config = memory_config.value_or(output_tensor.memory_config()),
+                },
+                ttnn::operations::normalization::SoftmaxDeviceOperation::tensor_args_t{
+                    .input_tensor = output_tensor,
+                }
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp:35]: instead of passing operation_args and tensor_args, this method accepts normal arguments and packs them to those args structures for internal use.

Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp:35]: in a way, they should not leak to an operation

File: ttnn/cpp/ttnn/operations/normalization/softmax/device/softmax_device_operation.hpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+#include "ttnn/decorators.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/compute_kernel_config.hpp"
+
+
+#include "softmax_types.hpp"
+
+namespace ttnn::operations::normalization {
+
+struct SoftmaxDeviceOperation {
+    struct operation_attributes_t {
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/normalization/softmax/device/softmax_device_operation.hpp:23]: In my PR I moved these args to a common place and completely moved out programs to their own .hpp and .cpp

File: ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/operations/transformer/attention_softmax/attention_softmax.cpp:35]: oh, at the same time

https://github.com/tenstorrent/tt-metal/pull/11953/files

**Review by yugaoTT — APPROVED** [review]
